tests/data_context/cloud_data_context/test_datasource_crud.py

```diff
@@ -0,0 +1,135 @@
+"""This file is meant for integration tests related to datasource CRUD."""
+import copy
+from unittest.mock import patch
+
+import pytest
+
+from great_expectations.data_context import BaseDataContext
+from great_expectations.data_context.store import GeCloudStoreBackend
+from great_expectations.data_context.types.base import DatasourceConfig
+from great_expectations.datasource import BaseDatasource
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "save_changes",
+    [
+        pytest.param(True, id="save_changes=True"),
+        pytest.param(False, id="save_changes=False"),
+    ],
+)
+@pytest.mark.parametrize(
+    "config_includes_name_setting",
+    [
+        pytest.param("name_supplied_separately", id="name supplied separately"),
+        pytest.param("config_includes_name", id="config includes name"),
+        pytest.param(
+            "name_supplied_separately_and_included_in_config",
+            id="name supplied separately and config includes name",
+            marks=pytest.mark.xfail(strict=True, raises=TypeError),
+        ),
+    ],
+)
+def test_cloud_mode_base_data_context_add_datasource(
```

Comment:
Note this uses a BaseDataContext. Similar tests will be coming that use DataContext and CloudDataContext.